org.eclipse.xtext/src/org/eclipse/xtext/xtext/ecoreInference/Xtext2EcoreInterpretationContext.java

```diff
@@ -190,8 +212,16 @@ public Xtext2EcoreInterpretationContext mergeSpawnedContexts(List<Xtext2EcoreInt
 		return result;
 	}
 
+	public Collection<EClassifierInfo> getCurrentTypes() {
+		return currentTypes;
+	}
+	
 	public EClassifierInfo getCurrentCompatibleType() {
-		return eClassifierInfos.getCompatibleTypeOf(currentTypes);
+		try {
+			return eClassifierInfos.getCompatibleTypeOf(currentTypes);
+		} catch(IllegalArgumentException e) {
```

Comment:
is this an expected case or an error case?

Reply:
It used to be there. I think it doesn't really happen in practice but I wanted to keep the catch clause.